skimage/filters/_gaussian.py

```diff
@@ -112,9 +112,17 @@ def gaussian(image, sigma=1, output=None, mode='nearest', cval=0,
             sigma = [sigma] * (image.ndim - 1)
         if len(sigma) != image.ndim:
             sigma = np.concatenate((np.asarray(sigma), [0]))
+    if output is not None:
+        if not isinstance(output, np.ndarray):
+            raise ValueError("Output is not an numpy array")
     image = convert_to_float(image, preserve_range)
-    return ndi.gaussian_filter(image, sigma, mode=mode, cval=cval,
-                               truncate=truncate)
+    if output:
+        if issubclass(image.dtype.type, output.dtype.type):
+            msg = ("Converted image dtype doesn't correspond output dtype, "
+                   "possible loss of data.")
+            warn(RuntimeWarning(msg))
+    return ndi.gaussian_filter(image, sigma, output=output, mode=mode,
```

Comment:
I would suggest something like this

```suggestion
    image = convert_to_float(image, preserve_range)
    if output is None:
        output = np.empty_like(image)
    elif not isinstance(output, np.ndarray):
        raise ValueError("provided output argument is not a numpy array")
    elif image.dtype != output.dtype:
        warn("Invalid output datatype. A new array is assigned.")
        output = np.empty_like(image)
    ndi.gaussian_filter(image, sigma, output=output, mode=mode,
                                   cval=cval, truncate=truncate)
    return output
```
In fact, with your implementation, if output is not None, ndi.gaussian_filter returns None

Reply:
My bad! ndi.gaussian_filter returns output even if it is  not None, sorry :-)

Reply:
@rfezzani `ndi.gaussian_filter` works great, even if `output` is not specified : ) `ndi.gaussian_filter` uses funcs, that just call `scipy`'s own `_ni_support._get_output`. That's why no need for `output = np.empty_like(image)`.

Reply:
If the `skimage.filters.gaussian`'s return is only floating data type, then your suggestion is ok, in this case it is needed to add in doc strings this requirement for output's dtype.  
But in case if we are saving `output`'s data type, then it looks like a bug.

Reply:
Since `skimage.filters.gaussian` uses the `preserve_range` argument as input, which is then provided to the `convert_t_float` function, I suppose that it returns only floating data types. And you are right, it must be in fact clearly  documented.

Reply:
Thank you for clarifying!
I was thinking that `gaussian` return's data type is the same as `output`'s, and actually if it's provided, then filtering result should be restored in `output`,  in other case image is converting to float and passing to `scipy.ndimage.gaussian_filter`. 
Existing `preserver_range` parameter does not give a clear understanding, that return type is floating. As the same as existing `cval` doesn't mean that `gaussian` uses only `constant` border mode, it will be ignored if `mode` is not `constant` (and this is documentated) :)